backend/api/Services/InspectionService.cs

```diff
@@ -142,11 +152,61 @@ private IQueryable<Inspection> GetInspections(bool readOnly = true)
                 return null;
             }
 
+            if (response.StatusCode != HttpStatusCode.OK)
+            {
+                logger.LogError("Could not get inspection data for inspection with Id {inspectionId}", inspectionId);
+                return null;
+            }
+
             var inspectionData = await response.Content.ReadFromJsonAsync<
                 IDAInspectionDataResponse
             >() ?? throw new JsonException("Failed to deserialize inspection data from IDA.");
 
             return inspectionData;
         }
+
+        public async Task<byte[]?> GetInspectionImage(string installationCode, string isarInspectionId)
+        {
+            Inspection? inspection;
+            try
+            {
+                inspection = await ReadByIsarInspectionId(isarInspectionId, readOnly: true);
+                if (inspection == null)
+                {
+                    logger.LogError($"Could not find inspection with isar inspection Id {isarInspectionId}.");
+                    return null;
+                }
+            }
+            catch (Exception e)
+            {
+                logger.LogError(e, $"Error while finding an inspection with isar inspection Id {isarInspectionId}. {e.Message}");
+                return null;
+            }
+
+            var inspectionData = await GetInspectionStorageInfo(inspection.IsarInspectionId);
+
+            if (inspectionData == null)
+            {
+                logger.LogError($"Could not find inspection data for inspection with isar Id {inspection.IsarInspectionId}.");
+                return null;
+            }
+
+            if (!inspectionData.BlobContainer.ToLower(CultureInfo.CurrentCulture).Equals(installationCode.ToLower(CultureInfo.CurrentCulture), StringComparison.Ordinal))
+            {
+                logger.LogError($"Could not find inspection data for inspection with isar Id {inspection.IsarInspectionId} because blob name {inspectionData.BlobName} does not match installation {installationCode}.");
```

Comment:
Do we compare name of the blob, or blob location?

Reply:
This is to make sure the blob container has the same name as the installation code, a bit reduntant but will help in case something goes wrong with the naming of the containers in the storage account. Agreed with Thomas it's worth keeping

Reply:
The log was wrong but the check was fine